02-peer-protocol.md

```diff
@@ -560,6 +1072,342 @@ ones it has sent so it can use them should they be requested by
+
+The sending node:
+  - MUST set `funding_feerate_perkw` to the feerate for this transaction
+  - If it requires the receiving node to only use confirmed inputs:
+    - MUST set `require_confirmed_inputs`
+
+The receiving node:
+  - MAY fail the negotiation if:
+    - the `locktime` is unacceptable
+    - the `funding_feerate_perkw` is unacceptable
+  - MUST fail the negotiation if:
+    - `require_confirmed_inputs` is set but it cannot provide confirmed inputs
+
+#### Rationale
+
+`temporary_channel_id` MUST be derived using a zeroed out basepoint for the
+peer's revocation basepoint. This allows the peer to return channel-assignable
+errors before the *accepter*'s revocation basepoint is known.
+
+`funding_feerate_perkw` indicates the fee rate that the opening node will
+pay for the funding transaction in satoshi per 1000-weight, as described
+in [BOLT-3, Appendix F](03-transactions.md#appendix-f-dual-funded-transaction-test-vectors).
+
+`locktime` is the locktime for the funding transaction.
+
+The receiving node, if the `locktime` or `funding_feerate_perkw` is considered
+out of an acceptable range, may fail the negotiation. However, it is
+recommended that the *accepter* permits the channel open to proceed
+without their participation in the channel's funding.
+
+Note that `open_channel`'s `channel_reserve_satoshi` has been omitted.
+Instead, the channel reserve is fixed at 1% of the total channel balance
+(`open_channel2`.`funding_satoshis` + `accept_channel2`.`funding_satoshis`)
+rounded down to the nearest whole satoshi or the `dust_limit_satoshis`,
+whichever is greater.
+
+Note that `push_msat` has been omitted.
+
+`second_per_commitment_point` is now sent here (as well as in `channel_ready`)
+as a convenience for implementations.
+
+The sending node may require the other participant to only use confirmed inputs.
+This ensures that the sending node doesn't end up paying the fees of a low
+feerate unconfirmed ancestor of one of the other participant's inputs.
+
+### The `accept_channel2` Message
+
+This message contains information about a node and indicates its
+acceptance of the new channel.
+
+1. type: 65 (`accept_channel2`)
+2. data:
+    * [`channel_id`:`temporary_channel_id`]
+    * [`u64`:`funding_satoshis`]
+    * [`u64`:`dust_limit_satoshis`]
+    * [`u64`:`max_htlc_value_in_flight_msat`]
+    * [`u64`:`htlc_minimum_msat`]
+    * [`u32`:`minimum_depth`]
+    * [`u16`:`to_self_delay`]
+    * [`u16`:`max_accepted_htlcs`]
+    * [`point`:`funding_pubkey`]
+    * [`point`:`revocation_basepoint`]
+    * [`point`:`payment_basepoint`]
+    * [`point`:`delayed_payment_basepoint`]
+    * [`point`:`htlc_basepoint`]
+    * [`point`:`first_per_commitment_point`]
+    * [`point`:`second_per_commitment_point`]
+    * [`accept_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `accept_tlvs`
+2. types:
+   1. type: 0 (`upfront_shutdown_script`)
+   2. data:
+       * [`...*byte`:`shutdown_scriptpubkey`]
+   1. type: 1 (`channel_type`)
```

Comment:
Any chance we can make `channel_type` a requirement for V2 channels now? Would be a good opportunity.

Reply:
I really don't think we should bundle features just because we want to force people to use an unrelated feature X to get feature Y. Dual funding does not depend in any way on channel types. If you want to make channel types mandatory, just make that feature mandatory on your node (which we can probably target for next releases, as every implementation has shipped support for channel types AFAIK?).

Reply:
Yeah makes sense. 👍

Reply:
> but if we do that we should definitely make it required here

I don't understand the reasoning here...If we make channel types mandatory in all implementations, we can then clean up the spec to make it mandatory in the spec as well (like we did with `var_onion_optin`). Why would we then need to create a dependency on it for dual funding? We'll make it mandatory in all cases, not specifically for dual funding?

Sorry if I'm missing something here, I can't see why would ever need to use feature dependencies for something else than a real logical dependency. Making a feature mandatory seems completely orthogonal to other features?